skypy/halo/tests/test_mass.py

```diff
@@ -1,26 +1,67 @@
 import numpy as np
-import scipy.stats
-import scipy.integrate
+from astropy.cosmology import Planck15
+from astropy.units import allclose
+from skypy.power_spectrum import _eisenstein_hu as eh
+from skypy.power_spectrum import growth_function
 
 
 import skypy.halo.mass as mass
 
+# Precomputed values for the test, for a Planck15 cosmology at redshift 0 and a
+# power spectrum given by the Eisenstein and Hu fitting formula
+# Models: Press-Schechter and Sheth-Tormen
+mass_array = [1.00000000e+10, 3.16227766e+10, 1.00000000e+11, 3.16227766e+11,
+              1.00000000e+12, 3.16227766e+12, 1.00000000e+13, 3.16227766e+13,
+              1.00000000e+14, 3.16227766e+14]
+PS_fsigma = [0.30839245, 0.3469052, 0.38721761, 0.42703783, 0.46176454,
+             0.48255685, 0.47362787, 0.41200955, 0.28215351, 0.11852313]
+ST_fsigma = [0.24531776, 0.26301646, 0.28088396, 0.29791293, 0.31217744,
+             0.32001412, 0.31466711, 0.28511577, 0.21918841, 0.1200341]
+mass_funct = [3.35678772e-11, 3.87001910e-12, 4.46652718e-13, 5.15069390e-14,
+              5.90581307e-15, 6.65419937e-16, 7.14923865e-17, 6.87084218e-18,
+              5.22044135e-19, 2.43974103e-20]
 
-def test_press_schechter():
-    # Test the schechter function, sampling dimensionless x values
-    n, m_star = 1, 1e9
-    alpha = - 0.5 * (n + 9.0) / (n + 3.0)
+cosmology = Planck15
+redshift = 0.0
+k = np.logspace(-3, 1, num=5, base=10.0)
```

Comment:
```suggestion
k = np.logspace(-3, 1, num=1000, base=10.0)
```
Some of your tests were failing because your power spectrum was too sparsely sampled and so the integral inside `_sigma_squared` was incorrectly evaluating to negative value at high mass and hence taking the square root gave a NaN. Increasing the resolution of the power spectrum as suggested should help. You should probably also do this in all your docstring tests as well.